internal/pkg/vm/actor/builtin/initactor/init.go

```diff
@@ -99,3 +138,190 @@ func (*Impl) GetNetwork(ctx runtime.Runtime) (string, uint8, error) {
 
 	return state.Network, 0, nil
 }
+
+// GetActorIDForAddress looks up the actor id for a filecoin address.
+func (a *Impl) GetActorIDForAddress(rt runtime.Runtime, addr address.Address) (types.Uint64, uint8, error) {
+	if err := rt.Charge(actor.DefaultGasCost); err != nil {
+		return 0, internal.ErrInsufficientGas, errors.RevertErrorWrap(err, "Insufficient gas")
+	}
+
+	var state State
+	err := actor.ReadState(rt, &state)
+	if err != nil {
+		return 0, errors.CodeError(err), err
+	}
+
+	ctx := context.TODO()
+	lookup, err := actor.LoadLookup(ctx, rt.Storage(), state.AddressMap)
+	if err != nil {
+		return 0, errors.CodeError(err), errors.RevertErrorWrapf(err, "could not load lookup for cid: %s", state.IdMap)
+	}
+
+	var id types.Uint64
+	err = lookup.Find(ctx, addr.String(), &id)
+	if err != nil {
+		if err == hamt.ErrNotFound {
+			return 0, errors.CodeError(err), errors.RevertErrorWrapf(err, "actor id not found for address: %s", addr.String())
+		}
+		return 0, errors.CodeError(err), errors.FaultErrorWrap(err, "could not lookup actor id")
+	}
+
+	return id, 0, nil
+}
+
+// GetAddressForActorID looks up the address for an actor id.
+func (a *Impl) GetAddressForActorID(rt runtime.Runtime, actorID types.Uint64) (address.Address, uint8, error) {
+	if err := rt.Charge(actor.DefaultGasCost); err != nil {
+		return address.Undef, internal.ErrInsufficientGas, errors.RevertErrorWrap(err, "Insufficient gas")
+	}
+
+	var state State
+	err := actor.ReadState(rt, &state)
+	if err != nil {
+		return address.Undef, errors.CodeError(err), err
+	}
+
+	ctx := context.TODO()
+	lookup, err := actor.LoadLookup(ctx, rt.Storage(), state.IdMap)
+	if err != nil {
+		return address.Undef, errors.CodeError(err), errors.RevertErrorWrapf(err, "could not load lookup for cid: %s", state.IdMap)
+	}
+
+	key, err := a.keyForActorID(actorID)
+	if err != nil {
+		return address.Undef, errors.CodeError(err), errors.FaultErrorWrapf(err, "could not encode actor id: %d", actorID)
+	}
+
+	var addr address.Address
+	err = lookup.Find(ctx, key, &addr)
```

Comment:
aside: The fact that the `LoadLookup` `Find` pattern is so common it might make sense to create a helper in actor/storage.go in the same style as `SetKeyValue`